### PR TITLE
Split GradleBuildConfigurationCacheSmokeTest and increase parallelism

### PR DESCRIPTION
**Pull request by blindpirate**
Let's split it and use bigger `maxParallelForks` given that we have better machines now.

### CODE REVIEW

**Comment by blindpirate** [comment]
@bot-gradle help

**Comment by blindpirate** [comment]
@bot-gradle test ast

**Comment by blindpirate** [comment]
@bot-gradle test ast

File: subprojects/smoke-test/build.gradle.kts [hunk 1]
```diff
@@ -84,7 +84,7 @@ tasks {
         group = "Verification"
         testClassesDirs = smokeTestSourceSet.output.classesDirs
         classpath = smokeTestSourceSet.runtimeClasspath
-        maxParallelForks = 1 // those tests are pretty expensive, we shouldn't execute them concurrently
+        maxParallelForks = 2 // those tests are pretty expensive, we shouldn't execute them concurrently
```
Comment (wolfs) [inline subprojects/smoke-test/build.gradle.kts:87]: I suppose the comment needs adjusting as well?

**Review by wolfs — APPROVED** [review]
LGTM!

You also said there were some problems with the JDK version picked up when running the embedded Gradle build. How did you fix those?

I don't see where you enable Test Distribution either.

**Comment by blindpirate** [comment]
@bot-gradle test and merge

File: subprojects/smoke-test/build.gradle.kts (hunk 1 above)
Comment (blindpirate) [reply subprojects/smoke-test/build.gradle.kts]: Done.

**Comment by blindpirate** [comment]
> You also said there were some problems with the JDK version picked up when running the embedded Gradle build. How did you fix those?

That would be in a followup PR.

> I don't see where you enable Test Distribution either.

I was trying to, but these smoke test uses so much memory that I have to configure a dedicated TD agent for them, otherwise the daemon are killed during the smoke tests. Now that it takes <20m on AMD machine, we no longer need TD.

**Comment by blindpirate** [comment]
@bot-gradle test and merge

**Pull request merged**